src/service/layers-impl.js

```diff
@@ -658,12 +755,66 @@ export class LayoutElement {
    * @return {!PositionDef}
    */
   getOffsetFromParent() {
-    if (this.needsRemeasure_) {
-      this.remeasure();
-    }
+    this.remeasure();
     return this.position_;
   }
 
+  /**
+   * Whether the layout is a descendant of the most recently scrolled layer.
+   *
+   * This MUST NOT be called unless inside an #ancestry iterator function.
+   * "Activeness" is only determined for the lifespan of the #ancestry call.
+   *
+   * @return {boolean}
+   */
+  isActive() {
+    return dev().assertBoolean(this.isActive_);
+  }
+
+  /**
+   * Gets the minimal horizontal distance of this element from it's parent's
+   * "viewport".
+   */
+  getHorizontalDistanceFromParent() {
+    const parent = this.getParentLayer();
+    if (!parent) {
+      return 0;
+    }
+
+    const {left} = this.getOffsetFromParent();
+    const {width} = this.getSize();
+    const scrollLeft = parent.getScrollLeft();
+    const parentWidth = parent.getSize().width;
+    const distance = Math.max(
+        0,
+        scrollLeft - (left + width),
+        left - (scrollLeft + parentWidth)
```

Comment:
Can we add inline comments to these two cases for clarity? E.g.

```
// Distance from left edge of parent to right edge of child.
// Distance from left edge of child to right edge of parent.
```

Or I think it would be even clearer if you extracted it into a couple of conditionals:

```
// Compared to parent's horizontal bounding box...
if (left + width < scrollLeft) { 
  // ...element does not intersect and is to the left.
  ...
} else if (left > scrollLeft + parentWidth) { 
  // ...element does not intersect and is to the right.
  ...
} else { 
  // ...element intersects it.
}
```